### PR TITLE
Extend `ID3D12DXVKInteropDevice` into `ID3D12DXVKInteropDevice1` to add `VkCommandBuffer` interop

### PR DESCRIPTION
**Pull request by liam-middlebrook**
Add functionality for grabbing the `VkCommandBuffer` which is internal to a `ID3D12CommandList` object. This allows for applications to do interop using DX12 synchronization primitives but enqueue work onto queues which are otherwise not exposed through traditional DX12 APIs.

This will be used as a basis for `dxvk-nvapi` to create a DX12 offering of the `VK_NV_optical_flow` functionality. Tagging @jp7677 and @Saancreed from that project as any feedback received here will be relevant to a follow-up pull request I will be making there.

### Interface Naming
Should this be called `ID3D12DXVKInteropDevice1` ? It has a high overlap in functionality with `ID3D12DXVKInteropDevice` so it seemed appropriate for it to inherit the methods. Saancreed suggested via Discord that it may make sense to rename `ID3D12DXVKInteropDevice` to `ID3D12VKInteropDevice` (keeping the same GUID, thus preserving ABI).

### Scope of VkCommandBuffer retrieval
I chose to keep a simpler implementation where we can only retrieve the internal `VkCommandBuffer` from command lists which are reset with an interop-enabled `ID3D12CommandAllocator`. We could also in theory define some more fine-grained synchronization contracts so that the internal command buffer could be retrieved from non-interop command lists.

I don't see a pressing use-case for that to be needed right now, but I can think of plenty reasons why it may be attractive though. One may be to move the VK calls for `CubinComputeShader*` functionality back into dxvk-nvapi, where it can just record dispatches into the command-buffer directly rather than relying on vkd3d-proton to be updated in order to support new functions in that space. I'm unclear on what the maintenance burden would be for supporting this kind of thing (I know that vkd3d-proton does commandlist-internal work tracking and this would definitely complicate that if it had to account for a mix of internally+externally recorded work).

### CODE REVIEW

File: libs/vkd3d/device_vkd3d_ext.c
Comment (HansKristian-Work) [inline libs/vkd3d/device_vkd3d_ext.c:422]: Bad intent.

File: libs/vkd3d/device_vkd3d_ext.c
Comment (HansKristian-Work) [inline libs/vkd3d/device_vkd3d_ext.c:516]: I guess using that bool to validate this is fine ...

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
The headers for commits should be vkd3d: not command:.

As for the name of the interface, I don't really care if it's ID3D12VKInterop or not, that's more up to @doitsujin.

**Comment by HansKristian-Work** [comment]
> I'm unclear on what the maintenance burden would be for supporting this kind of thing (I know that vkd3d-proton does commandlist-internal work tracking and this would definitely complicate that if it had to account for a mix of internally+externally recorded work).

I think this could work if there's a "begin interop" where we effectively invalidate all state in the command list (d3d12_invalidate_all_state), and flush any pending tracking. That basically means doing work that Close command list does.

File: libs/vkd3d/command.c
Comment (liam-middlebrook) [reply libs/vkd3d/command.c:2784]: The thought here was to specifically note whether or not a command allocator was going to be used for interop. So that the interop functions wouldn't be used with non-interop enabled allocators. This extra validation step isn't strictly necessary, so I'm happy to get rid of it if you prefer.

File: libs/vkd3d/command.c
Comment (liam-middlebrook) [reply libs/vkd3d/command.c:2825]: Thanks, that makes sense. That simplifies the change a lot as well.

File: libs/vkd3d/command.c
Comment (liam-middlebrook) [reply libs/vkd3d/command.c:2784]: On further thought (and reading your comments below) this is unnecessary. I think a model described below (with command list state invalidation) makes more sense, and is a bit more forward-thinking.

File: tests/d3d12_dxvk_interop_device.c
Comment (Saancreed) [inline tests/d3d12_dxvk_interop_device.c:104]: Shouldn't this now be something like the following?

```suggestion
    if (FAILED(hr = ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop(interop_device,
        (ID3D12CommandList*)command_list,
        &vkCmdBuf)))
    {
        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop failed %#x.\n", hr);
```

File: libs/vkd3d/device_vkd3d_ext.c
Comment (HansKristian-Work) [inline libs/vkd3d/device_vkd3d_ext.c:496]: This isn't quite enough to make the interop work, but it's a start. We can probably flesh that out after this lands. Just add a FIXME comment here for now.

File: tests/d3d12_dxvk_interop_device.c
Comment (HansKristian-Work) [inline tests/d3d12_dxvk_interop_device.c:100]: hr is not set here.

File: tests/d3d12_dxvk_interop_device.c
Comment (HansKristian-Work) [inline tests/d3d12_dxvk_interop_device.c:124]: This needs to be UINT64 to get the 8 byte alignment on 32-bit. Both the handle and offset it seems.

File: libs/vkd3d/breadcrumbs.c
Comment (HansKristian-Work) [inline libs/vkd3d/breadcrumbs.c:603]: Just keep this on one line, the indents here are a bit goofy. Ditto the similar code below.

File: include/vkd3d_device_vkd3d_ext.idl
Comment (HansKristian-Work) [inline include/vkd3d_device_vkd3d_ext.idl:69]: I think we need an EndVkCommandBufferInterop as well, just in case we need to suspect certain things in the future like conditional rendering and queries. For now, just implement it as a noop.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
The overall approach is reasonable I think and is not that invasive. Mostly nits now.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
This LGTM now, but needs to change the `command:` to `vkd3d:` on the commits as requested earlier.

**Review by HansKristian-Work — APPROVED** [review]
LGTM. Passes test suite on both NV and RADV.

**Pull request merged**
